### ISSUE TITLE
Dependency Detection error

### ISSUE THREAD

**Issue by christopherjbaker** [stealjs/steal#1388]
__How often can you reproduce it?__ 

- [X] Always
- [ ] Sometimes
- [ ] Rarely
- [ ] Unable
- [ ] I didn’t try

__Description:__

I encountered this when trying to import the `polished` package from npm, though I think I have identified the problem. I get the error `Uncaught Error: Module ./helpers/directionalProperty not declared as a dependency.` The line which has the require looks like this: `var _directionalProperty = /*#__PURE__*/require('./helpers/directionalProperty');`. I think the comment before `require(` is interfering with steal's dependency detection.

__Steps to reproduce:__

1. `npm install polished`
2. make a javascript file with `require('polished');`
3. execute said file through steal

__Environment:__

| Software                     | Version
| --------------------- | -------
| Steal version              | 1.11.5
| Steal-tools version    | 1.11.8
| node -v                       | 
| npm -v                        | 10.0.0
| Browser                      | chrome v66.0.3359.139
| Operating system      | macos 10.13.4

**Comment by James0x57** [comment]
Hm Steal is detecting it correctly using the contents of `polished/lib/index.js` directly... So I can't reproduce a parsing issue yet

<img src="https://user-images.githubusercontent.com/7545075/39578401-1468ca28-4eb2-11e8-8d59-f6c1dac9faa7.png" width="350px">

@christopherjbaker could you provide set up files for **Steps to reproduce** # 2 and 3 so I can take a closer look?

**Comment by matthewp** [comment]
Thanks! I was able to verify that removing the comment did fix the issue, so it's a bit weird. I think I can come up with a gist that reproduces.

**Comment by matthewp** [comment]
@James0x57 This gist recreates the issue: https://gist.github.com/matthewp/e8c088b27323a5c6f61d1b015fc15aa1

If you remove the comments it is able to load.

**Comment by matthewp** [comment]
Yeah, problem is definitely in the getCJSDeps() function, doesn't find the deps: 
<img width="515" alt="screen shot 2018-05-03 at 9 36 03 am" src="https://user-images.githubusercontent.com/361671/39579743-7eef48ec-4eb5-11e8-9a40-5a5bc1cf24fe.png">

**Comment by matthewp** [comment]
@James0x57 I think the problem is that the getCJSDeps function that you wrote is being used by the AMD extension, but an older / less accurate version is used by the CommonJS extension. I wonder if we used the same one if it would fix it.

**Comment by matthewp** [comment]
I've verified that using your getCJSDeps will fix it.

**Comment by James0x57** [comment]
Yep, I concluded the same thing. I never updated the CommonJS flavor of getCJSDeps. The AMD one works.
